sympy/functions/special/gamma_functions.py

```diff
@@ -319,6 +319,8 @@ def eval(cls, a, x):
 
         # Special values.
         if a.is_Number:
+            if a is S.Zero and not x is S.Zero:
+                return oo
```

Comment:
The test `not x is S.Zero` is insufficient because `x` could be an expression that is possibly equal to zero but not literally `S.Zero`.

Reply:
How do I tackle this ? Like do you mean to say I need to use something like `simplify/solve` (to simplify or solve the equation) here ?

Reply:
Definitely do not use `simplify` or `solve` as those are far too expensive. You could potentially use something like `x.is_zero is False` but even that can be expensive. It isn't necessary in all cases for the function to evaluate so you should consider whether it is even necessary to try and resolve any particular case.

Reply:
Actually I realize what we are trying to handle has already been taken care of , we can get rid of this check ! Cause the cases I was trying to tackle as being handled by lines (307-308) which go as follows
```
        if x is S.Zero:
            return S.Zero
```

Reply:
I'll try getting rid of that check cause the cases which I want to handle (when `x.is_number` and `not x.is_zero`) will always be handled correctly from the above reported line ! If you feel something should be reverted back , i'll do the needful @oscarbenjamin

Reply:
I see now that this is in an `a.is_Number` block. In that case we do know whether `a` is equal to zero so it's fine to check `a is S.Zero` (although I would write it as `a is not S.Zero`).

Reply:
Why do you prefer it as `a is not S.Zero` . I mean the special case goes as follows right ! 
1) if `a==0 `and `x == 0` : we should return `0`
2) else if `a== 0` and x is any other number : we should return `oo `

Like the special case is focused on the cases where `a== 0` , hence I don't see the negation case here , am I missing something . The first case is being answered on line 307 -308 and so I only need to address the second case which i've done in the latest commit !